### PR TITLE
IBX-8019: Replaced LocationService::loadLocationChildren usages with `SearchService::findLocations`

### PR DESCRIPTION
**Pull request by webhdx**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | [IBX-8019](https://issues.ibexa.co/browse/IBX-8019)
| **Type**                                   | improvement
| **Target Ibexa DXP version** | `v3.3`
| **BC breaks**                          | no
| **Doc needed**                       | no

Replaced usage of `LocationService::loadLocationChildren` in favor of `SearchService::findLocations` which operates much faster.

I wanted to keep changes as concise as possible, however found out that affected tests `LocationEventSubscriber` and `TrashEventSubscriber` did not checked run any checks against calling `EventNotificationService::sendNotification` which is what is the most important part of the subscribers. 

Related PRs:
- https://github.com/ezsystems/ezplatform-kernel/pull/407
- https://github.com/ezsystems/ezplatform-admin-ui/pull/2119
- https://github.com/ezsystems/ezrecommendation-client/pull/114

#### Checklist:
- [x] Provided PR description.
- [x] Tested the solution manually.
- [x] Provided automated test coverage.
- [x] Checked that target branch is set correctly (master for features, the oldest supported for bugs).
- [x] Ran PHP CS Fixer for new PHP code (use `$ composer fix-cs`).
- [x] Asked for a review (ping `@ezsystems/php-dev-team`).

### CODE REVIEW

File: tests/lib/Event/Subscriber/AbstractRepositoryEventSubscriberTest.php [hunk 1]
```diff
@@ -27,6 +29,12 @@ abstract class AbstractRepositoryEventSubscriberTest extends AbstractCoreEventSu
     /** @var \PHPUnit\Framework\MockObject\MockObject|\EzSystems\EzRecommendationClient\Helper\ContentHelper */
     protected $contentHelperMock;
 
+    /** @var \PHPUnit\Framework\MockObject\MockObject|\eZ\Publish\Core\Query\QueryFactoryInterface */
```
Comment (konradoboza) [inline tests/lib/Event/Subscriber/AbstractRepositoryEventSubscriberTest.php]:
```suggestion
    /** @var \PHPUnit\Framework\MockObject\MockObject&\eZ\Publish\Core\Query\QueryFactoryInterface */
```

File: tests/lib/Event/Subscriber/AbstractRepositoryEventSubscriberTest.php [hunk 2]
```diff
@@ -27,6 +29,12 @@ abstract class AbstractRepositoryEventSubscriberTest extends AbstractCoreEventSu
...
+    protected $queryFactoryMock;
+
+    /** @var \PHPUnit\Framework\MockObject\MockObject|\eZ\Publish\API\Repository\SearchService */
```
Comment (konradoboza) [inline tests/lib/Event/Subscriber/AbstractRepositoryEventSubscriberTest.php]:
```suggestion
    /** @var \PHPUnit\Framework\MockObject\MockObject&\eZ\Publish\API\Repository\SearchService */
```

File: tests/lib/Event/Subscriber/TrashEventSubscriberTest.php [hunk 3]
```diff
@@ -63,55 +67,124 @@ public function subscribedEventsDataProvider(): array
...
             ->method('loadContentInfo')
-            ->willReturn($this->contentInfo);
+            ->willReturnOnConsecutiveCalls(
+                $relationContentInfo1,
+                $relationContentInfo2,
+                $relationContentInfo3,
+            );
+
+        $this->notificationServiceMock
+            ->expects($this->exactly(4))
```
Comment (konradoboza) [inline tests/lib/Event/Subscriber/TrashEventSubscriberTest.php]:
```suggestion
            ->expects(self::exactly(4))
```

File: tests/lib/Event/Subscriber/TrashEventSubscriberTest.php [hunk 4]
```diff
@@ -63,55 +67,124 @@ public function subscribedEventsDataProvider(): array
...
             ->method('sudo')
-            ->with(static function () use ($contentInfo) {})
             ->willReturn($this->getRelationList());
 
-        $this->contentServiceMock
-            ->expects($this->atLeastOnce())
-            ->method('loadContentInfo')
-            ->willReturn($this->contentInfo);
+        $this->notificationServiceMock
+            ->expects($this->exactly(5))
```
Comment (konradoboza) [inline tests/lib/Event/Subscriber/TrashEventSubscriberTest.php]:
```suggestion
            ->expects(self::exactly(5))
```

**Review by Nattfarinn — APPROVED** [review]

**Comment by webhdx** [comment]
@konradoboza Fixed

**Review by konradoboza — APPROVED** [review]

**Comment by ciastektk** [comment]
@webhdx, CI is red

**Comment by webhdx** [comment]
@ciastektk Fixed, forgot to push updated PHPStan baseline 🤦

**Review by ciastektk — APPROVED** [review]

**Review by tomaszszopinski — APPROVED** [review]
QA approved on IbexaDXP exp 3.3.37.

**Pull request merged**